source/AI.cpp

```diff
@@ -659,7 +659,22 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			{
 				double health = .5 * it->Shields() + it->Hull();
 				double &threshold = appeasmentThreshold[it.get()];
-				if(1. - health > threshold)
+				// Get the list of hostile (enemy) in the system
+				const auto enemies = GetShipsList(*it, true);
+				// Check if any of them are currently capable of firing
+				// on this ship, with a slight buffer
+				bool dangerousEnemy = false;
+				for(const auto &enemy : enemies)
+				{
+					if(!enemy->IsDisabled() && enemy->GetTargetShip() == it)
+					{
+						dangerousEnemy = true;
+						break;
+					}
+				}
```

Comment:
Use `find_if`

```suggestion
				bool isTargeted = enemies.end() != find_if(enemies.begin(), enemies.end(), [&it](const Ship *foe)
						{ return !foe->IsDisabled() && foe->GetTargetShip() == it; });
```

I would also ask that this only be done when the threshold is guaranteed to be met, not prior to checking the threshold.

Reply:
I'm not sure yet how the `[&it](const Ship *foe)` gets the scope right inside the `find_if` (seems like `()` is reserved for the iterated item and `[]` is for others maybe?) but it looks fine otherwise and it compiles so it seems fine.  Also it looks like it's still outside the threshold check, but I'll merge your suggestion here and fix that after.

Reply:
https://en.cppreference.com/w/cpp/language/lambda